### PR TITLE
Bookstore: add reducers and actions

### PR DESCRIPTION
**Pull request by ocrearyy**
In this project, I did the following:

Add React Redux (npm install react-redux). ➕✔
Structure the application files using a "feature folder" approach and use the [ducks pattern](https://github.com/erikras/ducks-modular-redux) for your Redux files.🔗✔
Write the book's actions and reducer (in /src/redux/books/books.js) using the [ducks pattern](https://github.com/erikras/ducks-modular-redux).📙✔
Write the categories actions and reducer (in /src/redux/categories/categories.js) using the [ducks pattern](https://github.com/erikras/ducks-modular-redux).📁✔
Configure the Redux Store (/src/redux/configureStore.js):🏪✔

### CODE REVIEW

File: src/redux/books/books.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
+const ADD_BOOK = 'BOOK_ADDED';
+const REMOVE_BOOK = 'BOOK_REMOVED';
```
Comment (Meltrust) [inline src/redux/books/books.js]: - [x] These action types need to follow the duck's pattern, according to the project requirements:

image


Hint:  Examples of duck's pattern would be: 

```javascript
const ADD_BOOK = bookstore/books/ADD_BOOK 
const REMOVE_BOOK = bookstore/books/REMOVE_BOOK
```

File: src/redux/categories/categories.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+const CHECK_STATUS = 'CHECK_STATUS';
```
Comment (Meltrust) [inline src/redux/categories/categories.js]: - [x] These action types need to follow the duck's pattern, according to the project requirements:

image

Hint:  Examples of duck's pattern would be: 

```javascript
const CHECK_STATUS = bookstore/categories/CHECK_STATUS
```

**Review by Meltrust — CHANGES_REQUESTED** [review]
Hi @oshanedesign,

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



### To highlight:

- Nice first implementation and reducers✔️ 

You can do it

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/books/books.js (hunk 1 above)
Comment (ocrearyy) [reply src/redux/books/books.js]: @Meltrust  thank you for that. I have added it see below: 
e59

File: src/redux/categories/categories.js (hunk 2 above)
Comment (ocrearyy) [reply src/redux/categories/categories.js]: @Meltrust  also added here 
e60

**Review by Meltrust — APPROVED** [review]
Hi @oshanedesign,


## Wow, you did it 🎉
Brilliant
## Thank you for implementing the duck's pattern, as requested in the previous review 💪 🥇 ㊗️ 
### Well done!




Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**